scout/adapter/mongo/case.py

```diff
@@ -86,37 +86,32 @@ def _set_similar_phenotype_query(
             institute_id(str): institute to search cases for
         """
         hpo_terms = []
-        order = None
         CASE_SIMILAR_PROJECTION = {"phenotype_terms": 1}
+
         if query_field == "similar_case":
-            case_obj = self.case(
-                display_name=query_term,
-                institute_id=institute_id,
-                projection=CASE_SIMILAR_PROJECTION,
-            )
+            temp_query_or = {
+                "$or": [
+                    {"display_name": query_term},
+                    {"individuals.display_name": query_term},
+                    {"individuals.subject_id": query_term},
+                    {"_id": query_term},
+                ]
+            }
+            temp_query = {"$and": [{"owner": institute_id}, temp_query_or]}
+            case_obj = self.case_collection.find_one(temp_query, CASE_SIMILAR_PROJECTION)
             if case_obj is None:
                 query["_id"] = {"$in": []}  # No result should be returned by query
                 return
 
             for term in case_obj.get("phenotype_terms", []):
                 hpo_terms.append(term.get("phenotype_id"))
+
             similar_cases = self.cases_by_phenotype(hpo_terms, institute_id, case_obj["_id"])
         else:  # similar HPO terms
             hpo_terms = list(query_term.replace(" ", "").split(","))
             similar_cases = self.cases_by_phenotype(hpo_terms, institute_id, None)
 
-        if len(similar_cases) == 0:  # No cases similar to given phenotype
-            query["_id"] = {"$in": []}  # No result should be returned by query
-            return
-
-        similar_case_ids = []
-        order = []
-        for i in similar_cases:
-            similar_case_ids.append(i[0])
-            order.append(i[1])
```

Comment:
Order works, but applies to the order of similar cases only. It's a number that tells mongo how to sort the results. All good, but the problem is that the query for cases is runned another time later in the code and the case are returned in another (random) order, so if you order them by "order" you won't get the most similar cases on top.